src/main/java/com/employeesService/EmployeesService/service/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.employeesService.EmployeesService.service;
+
+import com.employeesService.EmployeesService.model.Department;
+import com.employeesService.EmployeesService.repository.DepartmentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DepartmentServiceImpl implements DepartmentService{
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    public DepartmentServiceImpl(DepartmentRepository departmentRepository){
+        this.departmentRepository=departmentRepository;
+    }
+
+    @Override
+    public Department save(Department department) {
+        return departmentRepository.save(department);
+    }
+
+    @Override
+    public List<Department> getAll() {
+        return departmentRepository.findAll();
+    }
+
+    @Override
+    public Department getById(String deptNo) {
+        Department department= departmentRepository.findById(deptNo).orElse(null);
+        return department;
+    }
+
+    @Override
+    public Department update(String deptNo,Department department) {
+        return departmentRepository.update(department);
+    }
```

Comment:
Did you test the case when there is no department with the given ID? The id passed as _path param_, not the one taken from the Request Body

Reply:
Don't forget this as well ^^